### PR TITLE
docs: fixing documentation

### LINKED ISSUE #834: Documentation links and formatting
While reading the documentation, I have been encountering what looks like what might be broken links or references, where there is a `{ref}` followed by what looks like should be a hyperlink of some sort

Example:
```
The conf file should follow the specifications as mentioned {ref}`here <conf_spec`.
```

Examples in code base:
* https://github.com/splunk/pytest-splunk-addon/blob/b789bb480e590c8182b94f35548ec4bcf7b3c263/docs/how_to_use.md?plain=1#L249
* https://github.com/splunk/pytest-splunk-addon/blob/b789bb480e590c8182b94f35548ec4bcf7b3c263/docs/index_time_tests.md?plain=1#L17
* https://github.com/splunk/pytest-splunk-addon/blob/b789bb480e590c8182b94f35548ec4bcf7b3c263/docs/index_time_tests.md?plain=1#L75

When rendered on the doc site they look like:

image

image

image

Should these be linking to something or formatting differently?

Also, anywhere there is an example cli argument, there is usually a placeholder explaining what the value should be, but they are all missing a closing `>`. 

Example: 
https://github.com/splunk/pytest-splunk-addon/blob/b789bb480e590c8182b94f35548ec4bcf7b3c263/docs/how_to_use.md?plain=1#L23

Should these instead be closed like below or is this intentional?

 ```bash
 pytest --splunk-type=external --splunk-app=<path-to-addon-package> --splunk-data-generator=<path to pytest-splunk-addon-data.conf file> --splunk-host=<hostname> --splunk-port=<splunk-management-port> --splunk-user=<username> --splunk-password=<password> --splunk-hec-token=<splunk_hec_token>
 ```

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
- Updated the md files for proper reference.
- Added missing brackets

Fixes #834

### CODE REVIEW

**Comment by kdoroszko-splunk** [comment]
Lets change 'is not be mapped' to 'is not mapped' in  these lines:  [cim_tests_line103](https://github.com/splunk/pytest-splunk-addon/pull/836/files#diff-5c33c54f1c36bbe05f2e1838b99f739313e466cc96ce28768f2eca6850c924cfR103), [cim_tests_line109](https://github.com/splunk/pytest-splunk-addon/pull/836/files#diff-5c33c54f1c36bbe05f2e1838b99f739313e466cc96ce28768f2eca6850c924cfR109)

**Comment by kdoroszko-splunk** [comment]
Should we consider updating https://github.com/splunk/pytest-splunk-addon/pull/836/files#diff-49ba940f564070d4fac730e4d91f02a138324367030b4d4cd653cdb807f35345R53 with the latest [template version](https://github.com/splunk/addonfactory-repository-template/blob/f40cc05596c54feea24988053d9dc4bbcb0130ac/enforce/docker-compose.yml#L53 ) (I mean remove volumes and adding platform for both services)?

**Comment by harshilgajera-crest** [comment]
> Should we consider updating https://github.com/splunk/pytest-splunk-addon/pull/836/files#diff-49ba940f564070d4fac730e4d91f02a138324367030b4d4cd653cdb807f35345R53 with the latest [template version](https://github.com/splunk/addonfactory-repository-template/blob/f40cc05596c54feea24988053d9dc4bbcb0130ac/enforce/docker-compose.yml#L53) (I mean remove volumes and adding platform for both services)?

Done

**Comment by harshilgajera-crest** [comment]
> Lets change 'is not be mapped' to 'is not mapped' in these lines: [cim_tests_line103](https://github.com/splunk/pytest-splunk-addon/pull/836/files#diff-5c33c54f1c36bbe05f2e1838b99f739313e466cc96ce28768f2eca6850c924cfR103), [cim_tests_line109](https://github.com/splunk/pytest-splunk-addon/pull/836/files#diff-5c33c54f1c36bbe05f2e1838b99f739313e466cc96ce28768f2eca6850c924cfR109)

Done

**Review by kdoroszko-splunk — DISMISSED** [review]

**Comment by kdoroszko-splunk** [comment]
https://github.com/splunk/pytest-splunk-addon/blob/b789bb480e590c8182b94f35548ec4bcf7b3c263/docs/generate_conf.md?plain=1#L48 - here are two missisng `>` sign

File: Dockerfile.tests [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-FROM ubuntu:latest
+FROM ubuntu:22.04
```
Comment (kdoroszko-splunk) [inline Dockerfile.tests:16]: Do we need that change?

Comment (artemrys) [reply Dockerfile.tests:16]: It did not work without it, let's leave it as it right now and then we can debug it.

**Review by kdoroszko-splunk — APPROVED** [review]

**Pull request merged**